### DISCUSSION TITLE
Run a similarity report against all my music to a file (prior to any importing) ?

### DISCUSSION THREAD

**Discussion by nodecentral** [beetbox/beets#4603] [Q&A]
Hi

I’ve finished importing a large chunk of my music which scored above 95% on the similarity rating, but before importing the rest I’d like to run a similarity report against everything left to see what they scored and where I need to focus my attention on increasing their %  (before running any import again ?)

**Comment by RollingStar** [comment]
You might get somewhere with

--pretend

--timid

and a few plugins: acoustid, fromfilename, discogs, and I think there's a barcode plugin as well.

You could import "asis" in quiet mode and then , I think, see what similarity score the imports got. Double check that before you go importing everything.

**Reply by nodecentral** [reply]
Many thanks for replying @RollingStar.

> To perform a “dry run”, just use the -p (for “pretend”) flag. This will show you a list of files that would be moved but won’t actually change anything on disk. The -t option sets the timid mode which will ask again before really moving or copying the files.

I had briefly tried `—pretend`, but it didn’t seem to do anything different ? 

What I’d love to see if a report run, that shows my album and then its closest match musicbrainz match (1st Candidate if multiple) with it’s % sent to a log file ..  Something like this (mock up below) would be amazing !!

````
downloads/Bob Dylan - Random Collection/Bob Dylan's Greatest Hits V2 (10 items)
https://musicbrainz.org/release/7ffd002d-3969-4835-a101-3e4d6d4237b5
(Similarity 44.0%) (missing tracks, tracks, year, ...) (CD, 1999, US, Columbia, CK 65975)

/downloads/Jack Johnson - On and On (15 items)
https://musicbrainz.org/release/7ffd002d-3969-4835-a101-3e4d6d4237b5
(Similarity: 91.3%) (tracks, missing tracks) (CD, 2003, US, Brushfire Records, 440 075 012 2)
````
Any ideas of how to get close to this sort of thing?

**Reply by RollingStar** [reply]
I looked through the code and `beet fields` and couldn't find anywhere where beets saves the match %. Maybe one of the more advanced devs can elaborate. Here are some starting points https://github.com/beetbox/beets/search?q=strong_rec_thresh&type=code

Maybe if you did pretend, and piped the log to a file >mylog.txt in verbose mode `-vv`? If you can edit the python source code you might be able to write whatever info you need to the console, even in 'quiet' (no user interaction) mode.

**Reply by sampsyo** [reply]
It's true that we don't currently save the match distance anywhere. That's an interesting idea!

One thing you could try, @nodecentral, is combining the importer's `quiet` mode with adjusting `strong_rec_thresh` such that *no* matches get automatically applied. In `quiet` mode, by default, anything that's not an "obviously" correct match just gets skipped. So if you turn that latter setting up high enough, *nothing* will be considered "obvious" and beets will just show the match and skip everything.

**Reply by nodecentral** [reply]
Thanks @sampsyo - what’s the best way to run that and have the results outputted to a text file for me to work on separately ?

**Reply by sampsyo** [reply]
You could consider using shell redirection, like `> mylog.txt 2>&1` or similar?
